tasks/pipeline.py

```diff
@@ -18,6 +18,8 @@
 
 # Tasks to trigger pipelines
 
+ALLOWED_REPO_BRANCHES = {"stable", "beta", "nightly", "none"}
```

Comment:
Does `none` have a special meaning? Where is it treated?

Reply:
Historically, setting the repo branch to `none` was the way to run a deploy pipeline without actually sending the artifacts anywhere. It's used in the `check_if_build_only` job:
https://github.com/DataDog/datadog-agent/blob/3eb5ceb4f7bba02da244b412bd1ac34e9367c967/.gitlab/check_deploy.yml#L49-L58

However, I'm not sure if that still has a purpose: maybe we can remove that option (and the associated job).